src/main/scala/models/Dialect.scala

```diff
@@ -0,0 +1,104 @@
+package CSVValidation
+
+import CSVValidation.traits.ObjectNodeExtentions.ObjectNodeGetMaybeNode
+import CSVValidation.traits.JavaIteratorExtensions.IteratorHasAsScalaArray
+
+import com.fasterxml.jackson.databind.node.{ArrayNode, ObjectNode, TextNode}
+
+object Dialect {
+
+  def fromJson(dialectNode: ObjectNode): Dialect = {
+    val encoding = dialectNode
+      .getMaybeNode("encoding")
+      .map(n => n.asText())
+      .getOrElse("UTF-8")
+
+    val lineTerminators = dialectNode
+      .getMaybeNode("lineTerminator")
+      .map {
+        case n: TextNode  => Array(n.asText())
+        case n: ArrayNode => n.iterator().asScalaArray.map(_.asText())
+        case n =>
+          throw MetadataError(s"Unexpected node type ${n.getClass.getName}")
+      }
+      .getOrElse(Array("\n", "\r\n"))
+
+    val quoteChar = dialectNode
+      .getMaybeNode("quoteChar")
+      .map(_.asText.charAt(0))
+      .getOrElse('\"')
+
+    val doubleQuote = dialectNode
+      .getMaybeNode("doubleQuote")
+      .map(_.asBoolean())
+      .getOrElse(true)
+
+    val commentPrefix = dialectNode
+      .getMaybeNode("commentPrefix")
+      .map(_.asText())
+      .getOrElse("#")
+
+    val header =
+      dialectNode.getMaybeNode("header").map(_.asBoolean()).getOrElse(true)
+
+    val headerRowCount =
+      dialectNode.getMaybeNode("headerRowCount").map(_.asInt).getOrElse(1)
+
+    val skipRows =
+      dialectNode.getMaybeNode("skipRows").map(_.asInt()).getOrElse(0)
+
+    val delimiter = dialectNode
+      .getMaybeNode("delimiter")
+      .map(_.asText().charAt(0))
+      .getOrElse(',')
+
+    val skipColumns =
```

Comment:
Are we using skipColumns anywhere?

Reply:
Not yet.

Reply:
The csvlint is not using it . I think our decision on the rest of the dialect stuff was to revisit them if needed by the tests,

Reply:
Can you add a new issue to look into it later? I really dislike the idea of us ignoring the spec in some places because it isn't tested. Let's make sure we know to fix it later.

Reply:
Okay, https://github.com/GSS-Cogs/csv-validation/issues/69